commons-numbers-gamma/src/main/java/org/apache/commons/numbers/gamma/RegularizedGamma.java

```diff
@@ -0,0 +1,210 @@
+                }
+            }
+        }
+    }
+
+    /**
+     * Creates the \( Q(a, x) \equiv 1 - P(a, x) \) <a href="http://mathworld.wolfram.com/RegularizedGammaFunction.html">
+     * regularized Gamma function</a>.
+     *
+     * Class is immutable.
+     */
+    public static final class Q {
+        /** Prevent instantiation. */
+        private Q() {}
+
+        /**
+         * Computes the regularized gamma function \( Q(a, x) = 1 - P(a, x) \).
+         *
+         * @param a Argument.
+         * @param x Argument.
+         * @return \( Q(a, x) \).
+         * @throws ArithmeticException if the continued fraction fails to converge.
+         */
+        public static double value(double a,
+                                   double x) {
+            return value(a, x, DEFAULT_EPSILON, Integer.MAX_VALUE);
+        }
+
+        /**
+         * Computes the regularized gamma function \( Q(a, x) = 1 - P(a, x) \).
+         *
+         * The implementation of this method is based on:
+         * <ul>
+         *  <li>
+         *   <a href="http://mathworld.wolfram.com/RegularizedGammaFunction.html">
+         *   Regularized Gamma Function</a>, equation (1).
+         *  </li>
+         *  <li>
+         *   <a href="http://functions.wolfram.com/GammaBetaErf/GammaRegularized/10/0003/">
+         *   Regularized incomplete gamma function: Continued fraction representations
+         *   (formula 06.08.10.0003)</a>
+         *  </li>
+         * </ul>
+         *
+         * @param a Argument.
+         * @param x Argument.
+         * @param epsilon Tolerance in continued fraction evaluation.
+         * @param maxIterations Maximum number of iterations in continued fraction evaluation.
+         * @throws ArithmeticException if the continued fraction fails to converge.
+         * @return \( Q(a, x) \).
+         */
+        public static double value(final double a,
+                                   double x,
+                                   double epsilon,
+                                   int maxIterations) {
+            if (Double.isNaN(a) ||
+                Double.isNaN(x) ||
+                a <= 0 ||
+                x < 0) {
+                return Double.NaN;
+            } else if (x == 0) {
+                return 1;
+            } else if (x < a + 1) {
+                // P should converge faster in this case.
+                return 1 - RegularizedGamma.P.value(a, x, epsilon, maxIterations);
+            } else {
+                final ContinuedFraction cf = new ContinuedFraction() {
+                        /** {@inheritDoc} */
+                        @Override
+                        protected double getA(int n, double x) {
+                            return n * (a - n);
+                        }
+
+                        /** {@inheritDoc} */
+                        @Override
+                        protected double getB(int n, double x) {
+                            return ((2 * n) + 1) - a + x;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of `Integer` multiplication is cast into a larger type. If you think the result of the multiplication could exceed the size of an `Integer`, consider casting the arguments of the multiplication to avoid arithmetic overflows. For more information, see [CWE-190](https://cwe.mitre.org/data/definitions/190.html).